src/cascadia/TerminalApp/TerminalPage.h

```diff
@@ -6,12 +6,108 @@
 #include "winrt/Microsoft.UI.Xaml.Controls.h"
 
 #include "TerminalPage.g.h"
+#include "Tab.h"
+#include "CascadiaSettings.h"
+#include "Profile.h"
+
+#include <winrt/Microsoft.Terminal.TerminalControl.h>
+#include <winrt/Microsoft.Terminal.TerminalConnection.h>
+#include <winrt/Microsoft.UI.Xaml.Controls.Primitives.h>
+#include <winrt/Windows.ApplicationModel.DataTransfer.h>
+#include <winrt/Microsoft.UI.Xaml.XamlTypeInfo.h>
 
 namespace winrt::TerminalApp::implementation
 {
     struct TerminalPage : TerminalPageT<TerminalPage>
     {
+    public:
         TerminalPage();
+
+        ~TerminalPage();
+
+        void SetSettings(::TerminalApp::CascadiaSettings* settings);
+
+        void Create();
+
+        hstring GetTitle();
+
+        void RefreshUIAfterSettingsReloaded();
+
+        void ShowOkDialog(const winrt::hstring& titleKey, const winrt::hstring& contentKey);
+
+        TerminalApp::TabRowControl* GetTabRow(); // This is a work-around because the winrt events are not working in Page
+
+        // -------------------------------- WinRT Events ---------------------------------
+        DECLARE_EVENT(TitleChanged, _titleChangeHandlers, winrt::Microsoft::Terminal::TerminalControl::TitleChangedEventArgs);
+        DECLARE_EVENT(LastTabClosed, _lastTabClosedHandlers, winrt::TerminalApp::LastTabClosedEventArgs);
+
+    private:
+        Microsoft::UI::Xaml::Controls::TabView _tabView{ nullptr };
+        TerminalApp::TabRowControl _tabRow{ nullptr };
+        Windows::UI::Xaml::Controls::Grid _tabContent{ nullptr };
+        Windows::UI::Xaml::Controls::SplitButton _newTabButton{ nullptr };
+
+        ::TerminalApp::CascadiaSettings* _settings;
```

Comment:
>; [](start = 50, length = 1)

Prefer initializing this raw pointer here with "{ nullptr }" rather than in the constructor.  (I wonder, though, if the various users should instead call a GetSettings function which will go retrieve it from the App.)

Reply:
Changed. 
For the second issue, I think maybe we need a static GetSettings method in App if we retrieve the settings from the app. TerminalPage is created by App so I think it is not good to call back to App.